RecyclerView/app/src/main/java/net/boondockradio/recyclerview/MainActivity.java

```diff
@@ -0,0 +1,85 @@
+package net.boondockradio.recyclerview;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+
+import java.util.ArrayList;
+
+public class MainActivity extends AppCompatActivity {
+
+    private ArrayList<String> states;
+    private RecyclerView recyclerView;
+    private RecyclerView.Adapter adapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        createDummyData();
+
+        recyclerView = (RecyclerView) findViewById(R.id.recycler_main_activity);
+        recyclerView.setHasFixedSize(true);
+        recyclerView.setLayoutManager(new LinearLayoutManager(this));
+
+        adapter = new StatesAdapter(states);
+        recyclerView.setAdapter(adapter);
```

Comment:
This is allowed as it is the specification in Java, how ever this way of initializing and setting member variable is uncommon.

I guess it is better to write this way ONLY IF all of your team members do like this and write this way and you are a kind of a person who likes codes to be consistent throughout your project.

Below is my personal opinion why I think you never should do this:

- much harder to read
  - inlined statement is not intuitively understandable
  - (I believe) almost no one does this